### PR TITLE
kubectl label command crashes when a resource has no labels

### PR DESCRIPTION
**Pull request by kazegusuri**
```
$ cat redis-master.json
{
  "id": "redis-master",
  "kind": "Pod",
  "apiVersion": "v1beta1",
  "desiredState": {
    "manifest": {
      "version": "v1beta1",
      "id": "redis-master",
      "containers": [{
        "name": "master",
        "image": "dockerfile/redis",
        "cpu": 100,
        "ports": [{
          "containerPort": 6379,
          "hostPort": 6379
        }]
      }]
    }
  }
}
$ kubectl create -f redis-master.json
$ kubectl get po
POD                 IP                  CONTAINER(S)        IMAGE(S)            HOST                        LABELS              STATUS
redis-master       10.244.23.2         master              dockerfile/redis    172.17.8.103/172.17.8.103   <none>              Running

$ kubectl label po redis-master name=bar 
# crash
```

### CODE REVIEW

**Comment by satnam6502** [comment]
I wonder if `api.ObjectMetaFor` should make sure it returns an empty map rather than nil rather than requiring this check for its callers?

**Comment by bgrant0607** [comment]
@brendandburns

**Comment by kazegusuri** [comment]
Modified `api.ObjectMetaFor` to returns an empty map.

**Comment by satnam6502** [comment]
Someone should check to ensure that this is a good change i.e. for `api.ObjectMetaFor` to return an empty map for empty labels rather than `nil`. @brendandburns or @bgrant0607 can probably make a pronouncement.

**Comment by roberthbailey** [comment]
@jlowdermilk reviewed ed9761f which originally added kubectl label and might also have an opinion.

**Comment by jlowdermilk** [comment]
If it's any guide, the [ObjectMeta](https://github.com/kazegusuri/kubernetes/blob/update_nil_label/pkg/api/types.go#L133) api object specifies labels as an omitempty field. I suspect there may be other code which assumes that objectmeta returns nil labels if the object has no labels. I recommend you just keep the original change in label.go.

File: pkg/api/meta.go [hunk 1]
```diff
@@ -46,5 +46,13 @@ func ObjectMetaFor(obj runtime.Object) (*ObjectMeta, error) {
 	if err := runtime.FieldPtr(v, "ObjectMeta", &objectMeta); err != nil {
 		return nil, err
 	}
+
+	if objectMeta.Labels == nil {
```
Comment (smarterclayton) [inline pkg/api/meta.go]: I'm uncomfortable with this - nil labels and nil annotations is a valid go-ism - making it here doesn't make this easier to consume.

**Comment by smarterclayton** [comment]
I would prefer clients simply guard against labels being nil.

**Comment by brendandburns** [comment]
Yeah, I think I'd prefer nil checks in the client also...  @kazegusuri thanks for the PR and sorry for the run around...  If you revert back to your original nil checks, I'll happily merge this PR.

Thanks again.

**Comment by kazegusuri** [comment]
OK, thank you for review. I reverted the second commit.

**Comment by brendandburns** [comment]
LGTM.  Merging.

(sorry for the delay!)

**Pull request merged**